solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -191,6 +201,68 @@ pub mod solana_ibc {
         chain.set_stake((validator).into(), amount)
     }
 
+    /// Sets up new fee collector proposal which wont be changed until the new fee collector
+    /// calls `accept_fee_collector_change`. If the method is called for the first time, the fee
+    /// collector would just be set without needing for any approval.
+    ///
+    pub fn setup_fee_collector<'a, 'info>(
+        ctx: Context<'a, 'a, 'a, 'info, SetupFeeCollector<'info>>,
+        new_fee_collector: Pubkey,
+    ) -> Result<()> {
+        let private_storage = &mut ctx.accounts.storage;
+
+        let signer = ctx.accounts.fee_collector.key();
+
+        if private_storage.fee_collector == Pubkey::default() {
+            private_storage.fee_collector = new_fee_collector;
+        } else if signer == private_storage.fee_collector {
+            private_storage.new_fee_collector_proposal = Some(new_fee_collector)
+        } else {
+            return Err(error::Error::InvalidFeeCollector)
+                .map_err(|err| error!((&err)));
```

Comment:
```suggestion
            return Err(error!(error::Error::InvalidFeeCollector));
```

should just work.  Here and below.

Reply:
Fixed in #293